test/wait_for_blocks.go

```diff
@@ -0,0 +1,74 @@
+package test
+
+import (
+	"context"
+
+	"golang.org/x/sync/errgroup"
+)
+
+// ChainHeighter fetches the current chain block height
+type ChainHeighter interface {
+	Height(ctx context.Context) (uint64, error)
+}
+
+// WaitForBlocks blocks until all chains reach a block height delta equal to or greater than the delta argument
+func WaitForBlocks(ctx context.Context, delta int, chains ...ChainHeighter) error {
+	var (
+		done  = make(chan struct{})
+		errCh = make(chan error)
+	)
+
+	var eg errgroup.Group
```

Comment:
I think this one would a proper use of https://pkg.go.dev/golang.org/x/sync/errgroup#WithContext -- if we are waiting for multiple chains, as soon as the first one fails, the shared context should be canceled so that we don't wait any longer for the other chains.